fusedrug/data/interaction/drug_target/datasets/dti_binding_dataset.py

```diff
@@ -112,7 +112,86 @@ def dti_binding_dataset(
     ]
     dynamic_pipeline = PipelineDefault("DTI dataset", dynamic_pipeline)
 
-    dataset = DatasetDefault(sample_ids=None, dynamic_pipeline=dynamic_pipeline)
+    dataset = DatasetDefault(
+        sample_ids=None, dynamic_pipeline=dynamic_pipeline
+    )  # TODO: sample_ids here should be either pairs_df.index, or len(pairs_df)
+    dataset.create()
+
+    return dataset
+
+
+def dti_binding_datase_combined(
```

Comment:
typo: dataset
Is it an alternative to the function above?
If yes, do we need to keep both of them?
Is it something you are calling to from bmfm-bench?

Reply:
Corrected the typo. It is indeed an alternative to the function above. They are located on different ends of the memory-speed tradeoff. The current one (dti-binding-dataset) is memory-light, but computationally intensive - we need to merge three dataframes in an operator called for every minibatch. The new one (dti_binding_dataset_combined) is quicker (no real-time merges during MB construction), but more memory-intensive, taking up x2-x3 more memory.
They are called from dataset-specific benchmarks, e.g. bmfm_bench/benchmarks/interaction/protein_protein_interaction/benchmark.py

Reply:
So are they both in fuse-drug? They don't seem like something you can reuse.
I think we should put them in bmfm-bench.

Reply:
They are currently used by both PPI and DTI benchmarks, and by any affinity dataset we ingest using the current interaction benchmarks. I do think the code could use some cleanup - I'm not sure it's all being used, and maybe it is better off in benchmarks. We should take this up with Alex - he's much more familiar with the code. I suggest you open an issue.

Reply:
@alex-golts . what is your take on this? Should we move it to bmfm-bench?

Reply:
@mosheraboh Currently the primary use of these functions is via benchmarks (since bmfm_bench started to require benchmarks to implement a PyTorch dataset), so I can see why it can make sense to move them to bmfm_bench.
But that can change (for example if/when we publish more DTI/PPI model examples).

Why do you think these functions "don't seem like something you can reuse"?
at some point in the past, we specifically completed a task to make the pre-existing dti_binding_dataset in "fuse-style".